src/hooks/useAmp.ts

```diff
@@ -0,0 +1,21 @@
+import dispatcher from '../dispatcher';
+import { isServerSide, ampScriptSrc } from '../utils';
+import { useEffect } from 'react';
+
+export const useAmp = () => {
+  if (isServerSide) {
+    dispatcher._setAmp();
+  }
+
+  useEffect(() => {
+    const nodeList = document.querySelectorAll(`script[src="${ampScriptSrc}"]`);
```

Comment:
Not currently allowed, but there are plans to allow hosts to self-host AMP scripts. In the future this might need modification to support this pattern (No need to change anything right now).